waspc/data/Generator/templates/react-app/src/queries/core.d.ts

```diff
@@ -1,3 +1,13 @@
-import { type Query } from './index'
+import { Query } from '.'
+import { Expand, _Awaited } from '../universal/types'
 
-export function createQuery<Input, Output>(queryRoute: string, entitiesUsed: any[]): Query<Input, Output>
+export function createQuery<BackendQuery extends GenericBackendQuery>(
+  queryRoute: string,
+  entitiesUsed: any[]
+): QueryFor<BackendQuery>
+
+type QueryFor<BackendQuery extends GenericBackendQuery> = Expand<
+  Query<Parameters<BackendQuery>[0], _Awaited<ReturnType<BackendQuery>>>
+>
```

Comment:
Applying `Expand` to this type results in a better DX.

On-hover with `Expand`:

![image](https://user-images.githubusercontent.com/22752307/230371004-ac6a6eb9-4518-408d-ac1d-ed5287509dc6.png)

On-hover without `Expand`: 

![image](https://user-images.githubusercontent.com/22752307/230371331-71a77b97-a55c-4296-ae6d-22a8cae30e7e.png)